versions.tf

```diff
@@ -22,3 +26,18 @@ provider "aws" {
     tags = var.tags
   }
 }
+
+
+provider "aws" {
+  alias  = "eks"
+  region = var.region
+  default_tags {
+    tags = var.tags
+  }
+
+  assume_role {
+    # https://github.com/hashicorp/terraform/issues/30690
+    # https://github.com/hashicorp/terraform/issues/2430
+    role_arn = "${aws_iam_role.create_eks_role.arn}${time_sleep.create_eks_role_10_seconds.id == "nil" ? "" : ""}"
```

Comment:
this is just another highlight that we should be building this as separate TF steps.